README.md

```diff
@@ -32,31 +32,128 @@ Starknet Foundry, like its [Ethereum counterpart](https://github.com/foundry-rs/
 
 ## Installation
 
-To install Starknet Foundry, first install `snfoundryup` by running:
+### Linux and MacOS
+
+#### Install Rust >= 1.80.1
+
+```shell
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+To verify that correct Rust version was installed, run
+
+```shell
+rustc --version
+```
+
+See [Rust docs](https://doc.rust-lang.org/beta/book/ch01-01-installation.html#installation) for more details.
+
+#### Install ASDF
+
+Follow the instructions from [asdf docs](https://asdf-vm.com/guide/getting-started.html#getting-started).
+
+To verify that ASDF was correctly installed, run
+
+```shell
+asdf --version
+```
+
+#### Install Scarb
+
+First, add Scarb plugin to ASDF
+
+```shell
+asdf plugin add scarb
+```
+
+Install Scarb
 
 ```shell
-$ curl -L https://raw.githubusercontent.com/foundry-rs/starknet-foundry/master/scripts/install.sh | sh
+asdf install scarb latest
 ```
 
-Follow the instructions and then run
+To verify that Scarb was correctly installed, run
 
 ```shell
-$ snfoundryup
+scarb --version
 ```
 
-You can also specify a version you wish to install:
+#### Install Starknet Foundry
+
+First, add Starknet Foundry plugin to ASDF
 
 ```shell
-$ snfoundryup -v 0.9.0
+asdf plugin add starknet-foundry
 ```
 
-To verify that the Starknet Foundry is installed correctly, run `snforge --version` and `sncast --version`.
+Install Starknet Foundry
+
+```shell
+asdf install starknet-foundry latest
+```
 
-## Features
+To verify that Starknet Foundry was correctly installed, run
+
+```shell
+snforge --version
+```
 
-- Fast testing framework `snforge` written in Rust
-- High-quality dependency management using [scarb](https://github.com/software-mansion/scarb)
-- Intuitive interactions and deployment of Starknet contracts through `sncast`
+or
+
+```shell
+sncast --version
+```
+
+### Windows
+
+#### Install Rust >= 1.80.1
+
+Go to https://www.rust-lang.org/tools/install and follow the installation instructions.
+
+To verify that correct Rust version was installed, run
+
+```shell
+rustc --version
+```
+
+See [Rust docs](https://doc.rust-lang.org/beta/book/ch01-01-installation.html#installation) for more details.
+
+#### Install Scarb
+
+1. Download the release archive matching your CPU architecture
+   from https://docs.swmansion.com/scarb/download.html#precompiled-packages.
+2. Extract it to a location where you would like to have Scarb installed. We recommend `%LOCALAPPDATA%\Programs\scarb`.
+3. From this directory, get the full path to `scarb\bin` and add it to PATH.
+   See [this article](https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/) for instructions on
+   Windows 10 and 11.
+
+To verify that Scarb was correctly installed, run
+
+```shell
+scarb --version
+```
```

Comment:
Maybe add link to scarb docs?  https://docs.swmansion.com/scarb/download.html#windows